ghost/core/core/server/data/migrations/versions/5.14/2022-09-02-12-55-rename-members-bio-to-expertise.js

```diff
@@ -1,34 +1,54 @@
 const logging = require('@tryghost/logging');
 const {createNonTransactionalMigration} = require('../../utils');
+const DatabaseInfo = require('@tryghost/database-info');
 
-module.exports = createNonTransactionalMigration (
+module.exports = createNonTransactionalMigration(
     async function up(knex) {
-        // check if the column exists before trying to rename it
+        logging.info('Renaming members.bio to members.expertise');
+
         const hasBio = await knex.schema.hasColumn('members', 'bio');
+        if (!hasBio) {
+            logging.warn('members.bio does not exist, skipping rename');
+            return;
+        }
+
         const hasExpertise = await knex.schema.hasColumn('members', 'expertise');
+        if (hasExpertise) {
```

Comment:
Would we want to throw here? Or drop the `bio` column maybe?

Having both a `bio` and `expertise` column is a completely invalid state

Reply:
Good question... I think we should throw so as to gracefully bail out and avoid further damage

Reply:
That said, migrations should be idempotent so if it already exists, we shouldn't try and do anything with it

I'm thinking I should collapse it back down into one if-statement and only do the rename if it's in the valid state

Reply:
Ok how's this? (latest push)